ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java

```diff
@@ -17,23 +19,51 @@
 
 public class JailPanel extends ChessPanel {
 
-    private final int mJailDimension;
+    private final int JAIL_WIDTH = 4;
     private final List<SquareJLabel> mSquareLabels;
+    private final BoardSize mJailSize;
 
     public JailPanel(int totalPieces) {
         super(false);
 
-        mJailDimension = (totalPieces / 4) + (totalPieces % 4);
+        if(totalPieces % 4 == 0) {
```

Comment:
Checking the Mod to see if there is remainder or not. If there is remainder we add an additional row of 4.